File: cpp/0541.0_Reverse_String_II.cpp
/*
执行用时：4 ms, 在所有 C++ 提交中击败了85.41% 的用户
内存消耗：7.2 MB, 在所有 C++ 提交中击败了41.52% 的用户
*/
class Solution {
public:
    string reverseStr(string s, int k) {
        int n = s.length();
        for (int i = 0; i < n; i += 2 * k) {
            reverse(s.begin() + i, s.begin() + min(i + k, n));
        }
        return s;
    }
};

File: go/0541.0_Reverse_String_II.go
/*

执行用时：0 ms, 在所有 Go 提交中击败了100.00% 的用户
内存消耗：3.4 MB, 在所有 Go 提交中击败了100.00% 的用户
*/

func reverseStr(s string, k int) string {
    t := []byte(s)
    for i := 0; i < len(s); i += 2 * k {
        sub := t[i: min(i + k, len(s))]
        for j, n := 0, len(sub); j < n / 2; j++ {
            sub[j], sub[n - 1 - j] = sub[n - 1 - j], sub[j]
        }
    }
    return string(t)
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}
